pkg/resourcemanager/psql/psqluser/psqluser.go

```diff
@@ -0,0 +1,285 @@
+
+	db, err := sql.Open(drivername, connString)
+	if err != nil {
+		return db, err
+	}
+
+	err = db.PingContext(ctx)
+	if err != nil {
+		return db, err
+	}
+
+	return db, err
+}
+
+// GrantUserRoles grants roles to a user for a given database
+func (s *PostgreSqlUserManager) GrantUserRoles(ctx context.Context, user string, roles []string, db *sql.DB) error {
+	var errorStrings []string
+	for _, role := range roles {
+		tsql := fmt.Sprintf("GRANT %s TO %q", role, user)
+
+		_, err := db.ExecContext(ctx, tsql)
+		if err != nil {
+			errorStrings = append(errorStrings, err.Error())
+		}
+	}
+
+	if len(errorStrings) != 0 {
+		return fmt.Errorf(strings.Join(errorStrings, "\n"))
+	}
+	return nil
+}
+
+// CreateUser creates user with secret credentials
+func (s *PostgreSqlUserManager) CreateUser(ctx context.Context, secret map[string][]byte, db *sql.DB) (string, error) {
+	newUser := string(secret[PSecretUsernameKey])
+	newPassword := string(secret[PSecretPasswordKey])
+
+	// make an effort to prevent sql injection
+	if err := findBadChars(newUser); err != nil {
+		return "", fmt.Errorf("Problem found with username: %v", err)
+	}
+	if err := findBadChars(newPassword); err != nil {
+		return "", fmt.Errorf("Problem found with password: %v", err)
+	}
+
+	tsql := fmt.Sprintf("CREATE USER \"%s\" WITH PASSWORD '%s'", newUser, newPassword)
+	_, err := db.ExecContext(ctx, tsql)
+
+	// TODO: Have db lib do string interpolation
+	//tsql := fmt.Sprintf(`CREATE USER @User WITH PASSWORD='@Password'`)
+	//_, err := db.ExecContext(ctx, tsql, sql.Named("User", newUser), sql.Named("Password", newPassword))
+
+	if err != nil {
+		return newUser, err
+	}
+	return newUser, nil
+}
+
+// UpdateUser - Updates user password
+func (s *PostgreSqlUserManager) UpdateUser(ctx context.Context, secret map[string][]byte, db *sql.DB) error {
+	user := string(secret[PSecretUsernameKey])
+	newPassword := helpers.NewPassword()
+
+	// make an effort to prevent sql injection
+	if err := findBadChars(user); err != nil {
+		return fmt.Errorf("Problem found with username: %v", err)
+	}
+	if err := findBadChars(newPassword); err != nil {
+		return fmt.Errorf("Problem found with password: %v", err)
+	}
+
+	tsql := fmt.Sprintf("ALTER USER '%s' WITH PASSWORD '%s'", user, newPassword)
+	_, err := db.ExecContext(ctx, tsql)
+
+	return err
+}
+
+// UserExists checks if db contains user
+func (s *PostgreSqlUserManager) UserExists(ctx context.Context, db *sql.DB, username string) (bool, error) {
+
+	tsql := fmt.Sprintf("SELECT * FROM pg_user WHERE usename = '%s'", username)
+	res, err := db.ExecContext(ctx, tsql)
+	if err != nil {
+		return false, err
+	}
+	rows, err := res.RowsAffected()
+	return rows > 0, err
+
+}
+
+// DropUser drops a user from db
+func (s *PostgreSqlUserManager) DropUser(ctx context.Context, db *sql.DB, user string) error {
+	tsql := fmt.Sprintf("DROP USER IF EXISTS %q", user)
```

Comment:
you will want to do the filtering on the user variable here too

Reply:
@frodopwns  Erin, this cannot use the SQL filtering as the username needs to be double quoted in this command if there is any special char inside, like '-' etc.